modules/router-store/src/router_store_module.ts

```diff
@@ -118,6 +175,14 @@ export function routerReducer<T = SerializedRouterStateSnapshot>(
 
 export interface StoreRouterConfig {
   stateKey?: string;
+  serializer?: new () => RouterStateSerializer;
+  /**
+   * By default, ROUTER_NAVIGATION is dispatched before guards and resolvers run.
+   * Therefore, the action could run too soon, for example
+   * there may be a navigation cancel due to a guard saying the navigation is not allowed.
+   * To run ROUTER_NAVIGATION after guards and resolvers, set this property to true.
+   */
+  dispatchNavActionOnEnd?: boolean;
```

Comment:
I think this should be more flexible. 

```
enum NavigationAction {
  PreActivation = 1,
  PostActivation = 2
};

navigationAction?: NavigationAction
```

The default would be PreActivation